atomic_reactor/plugins/pre_koji_parent.py

```diff
@@ -216,22 +220,35 @@ def manifest_list_entries_match(self, image, build_id):
                 return False
 
             arch = manifest['platform']['architecture']
+            if arch not in self.architectures:
+                self.log.warning('Requested platform %s is missing in manifest list', arch)
+                return False
             v2_digest = manifest['digest']
             manifest_list_data[arch] = v2_digest
 
         archives = self.koji_session.listArchives(build_id)
         koji_archives_data = {}
         for archive in (a for a in archives if a['btype'] == KOJI_BTYPE_IMAGE):
             arch = archive['extra']['docker']['config']['architecture']
+            if arch not in self.architectures:
+                self.log.warning('Requested platform %s is missing in Koji archive', arch)
+                return False
             v2_digest = archive['extra']['docker']['digests'][v2_type]
             koji_archives_data[arch] = v2_digest
 
-        if koji_archives_data == manifest_list_data:
-            self.log.info('Deeper manifest list check verified v2 manifest references match')
-            return True
-        self.log.warning('Manifest list refs "%s" do not match koji archive refs "%s"',
-                         manifest_list_data, koji_archives_data)
-        return False
+        # manifest lists can be manually pushed to the registry to make sure a specific tag
+        # (e.g., latest) is available for all platforms.
+        # In such cases these manifest lists may include images from different koji builds.
+        # We only want to check the digests for the images built in the current parent koji build
+        for arch in self.architectures:
+            if manifest_list_data(arch) != koji_archives_data(arch):
```

Comment:
we could also first check for which arches digests don't match, 
and then return only those arches for which it didn't match

Reply:
This?
```
        msg = 'Manifest list digest {} differs from Koji archive digest {} for platform {},'
        error_msg = ''
        for arch in self.architectures:
            if manifest_list_data(arch) != koji_archives_data(arch):
                error_msg += msg.format(manifest_list_data(arch), koji_archives_data(arch), arch)

        if error_msg:
            self.log.warning(error_msg)
            return False

```